### PR TITLE
Chapter 10

### PR DESCRIPTION
**Pull request by aisharina99**
ready

### CODE REVIEW

**Comment by aisharina99** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -5,10 +5,9 @@ class User < ApplicationRecord
   validates :name, presence: true
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
```
Comment (huongnt-2545) [inline app/models/user.rb]:
```suggestion
  validates :email, presence: true, uniqueness: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
```

File: app/views/shared/_error_messages.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<% if object.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains
+      <%= pluralize(@user.errors.count, "error") %>.
```
Comment (huongnt-2545) [inline app/views/shared/_error_messages.html.erb]:
```suggestion
      <%= pluralize(object.errors.count, "error") %>.
```

File: app/views/shared/_error_messages.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    </div>
+    <ul>
+    <% @user.errors.full_messages.each do |msg| %>
```
Comment (huongnt-2545) [inline app/views/shared/_error_messages.html.erb]:
```suggestion
    <% object.errors.full_messages.each do |msg| %>
```

**Comment by aisharina99** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged
You can merge this pull

**Pull request merged**